src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue47/SetWidthTest.kt

```diff
@@ -0,0 +1,78 @@
+package it.krzeminski.snakeyaml.engine.kmp.issues.issue47
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import it.krzeminski.snakeyaml.engine.kmp.api.Dump
+import it.krzeminski.snakeyaml.engine.kmp.api.DumpSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.Load
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.StringStreamDataWriter
+import it.krzeminski.snakeyaml.engine.kmp.common.ScalarStyle
+import it.krzeminski.snakeyaml.engine.kmp.emitter.Emitter
+import it.krzeminski.snakeyaml.engine.kmp.events.*
+
+class SetWidthTest : FunSpec({
+    val stringToSerialize = "arn:aws:iam::12345678901234567890:foobarbaz:testing:testing2:role/github-actions-role/\${{ github.token }}"
+
+    test("emit plain string and split") {
+        val settings = DumpSettings.builder().setWidth(80).build() // Intentionally limited.
+        val writer = StringStreamDataWriter()
+        val dump = Dump(settings)
+        dump.dump(stringToSerialize, writer)
+        val yaml = writer.toString()
+        val expected = "arn:aws:iam::12345678901234567890:foobarbaz:testing:testing2:role/github-actions-role/\${{\n  github.token }}"
+        parseBack(yaml) shouldBe stringToSerialize
+        yaml.trim() shouldBe expected
+    }
+
+    test("emit plain and split") {
+        val settings = DumpSettings.builder().setWidth(80).build() // Intentionally limited.
+        val writer = StringStreamDataWriter()
+        with(Emitter(settings, writer)) {
+            emit(StreamStartEvent())
+            emit(DocumentStartEvent(false, null, emptyMap()))
+            emit(ScalarEvent(null, null, ImplicitTuple(true, true), stringToSerialize, ScalarStyle.PLAIN))
+            emit(DocumentEndEvent(false))
```

Comment:
note: It is hard to understand what these `null` and `false` parameters mean.

suggestion: we could either use names for these arguments or (I would prefer this option) we could extract creation of these event objects into methods (if I am not mistaken only ScalarStyle is being changed) and use meaningful name for those methods (e.g `scalarWithStyle(style)`)

Reply:
Fair point as well, let's use named arguments: https://github.com/krzema12/snakeyaml-engine-kmp/pull/517/commits/f2e8d045a8bc63526b9bc4c25a902d965459865e